enhancements/network/ip-interface-selection.md

```diff
@@ -0,0 +1,285 @@
+---
+title: ip-interface-selection
+authors:
+  - @cybertron
+reviewers:
+  - @jcaamano
+  - @tsorya
+  - @flaper87
+approvers:
+  - @danwinship
+  - @trozet
+api-approvers:
+  - "None"
+creation-date: 2022-07-07
+last-updated: 2022-07-29
+tracking-link:
+  - https://issues.redhat.com/browse/OPNET-134
+see-also:
+  - https://github.com/openshift/baremetal-runtimecfg/issues/119
+replaces:
+superseded-by:
+---
+
+# Host IP and Interface Selection
+
+## Summary
+
+As OpenShift is deployed in increasingly complex networking environments, we
+have gotten many requests for more control over which interface is used for
+the primary node IP. We provided a basic mechanism for this with
+[KUBELET_NODEIP_HINT](https://github.com/openshift/machine-config-operator/pull/2888)
+but as users have started to exercise that some significant limitations have
+come to light.
+
+## Motivation
+
+Some users want to have a great deal of control over how their network traffic
+is routed. Because we use the default route for interface and IP selection,
+in some cases they are not able to route traffic the way they want.
+
+### User Stories
+
+As a deployer, I want my cluster traffic to stay on an isolated network with
+no external gateway. External traffic will travel on a different interface
+that is managed by a service like MetalLB.
```

Comment:
Would there be this other possibility?
* running kubelet on one interface
* running the pod network on a different interface

I don't know if if it is reasonable at all, probably does not make a lot of sense. But in that it does we would not want to use nodeip-configuration decision I guess.

Reply:
Yeah, my thought is that for the purposes of this enhancement we make sure everything is consistently picking the same interface, then as a followup we come up with a more complete solution that allows deployers to pick what they want to run where. I think that's going to be a lot harder to design so I think it's worth doing it in two steps.